project-ideas/order-ledger/createSalesOrder.md

```diff
@@ -1,47 +1,210 @@
+    "orderName": "#499999",
+    "orderTypeId": "SALES_ORDER",
+    "statusId": "ORDER_CREATED",
+    "originFacilityId": "",
+    "presentmentCurrencyCode": "USD",
+    "priority": "2",
+    "productStoreId": "STORE",
+    "salesChannelEnumId": "UNKNWN_SALES_CHANNEL",
+    "webSiteId": "WEBSTORE",
+    "adjustments": [
+        {
+            "orderAdjustmentTypeId": "SHIPPING_CHARGES",
+            "amount": "500",
+            "exemptAmount": "2.5",
+            "comments": "",
+            "sourcePercentage": "5",
+            "adjustmentAttributes": [
+                {
+                    "attrName": "taxAuthority",
+                    "attrValue": "State of California",
+                    "attrDescription": "state tax"
+                }
+            ]
+        }
+    ],
+    "attributes": [
+        {
+            "attrName": "item",
+            "attrValue": "testing attribute",
+            "attrDescription": ""
+        }
+    ],
+    "email": "abc@gmail.com",
+    "customer": {
+        "partyId": "Doe",
+        "externalId": 22889483207024,
+        "first_name": "John",
+        "last_name": "Doe"
+    },
+    "billing_address": {
+        "email": "abc@gmail.com",
+        "first_name": "John",
+        "address1": "1020 Thalia Street",
+        "phone": "+17869573716",
+        "city": "New Orleans",
+        "zip": "70130",
+        "province": "Louisiana",
+        "country": "United States",
+        "last_name": "Doe",
+        "address2": null,
+        "company": null,
+        "latitude": 29.9377606,
+        "longitude": -90.07084449999999,
+        "name": "John Doe",
+        "country_code": "US",
+        "province_code": "LA"
+    },
+    "shipping_address": {
+        "email": "abc@gmail.com",
+        "first_name": "John",
+        "address1": "1020 Thalia Street",
+        "phone": "+17869573716",
+        "city": "New Orleans",
+        "zip": "70130",
+        "province": "Louisiana",
+        "country": "United States",
+        "last_name": "Doe",
+        "address2": null,
+        "company": null,
+        "latitude": 29.9377606,
+        "longitude": -90.07084449999999,
+        "name": "John Doe",
+        "country_code": "US",
+        "province_code": "LA"
+    },
+    "identifications": [
+        {
+            "idType": "SHOPIFY_ORD_NAME",
+            "idValue": "2345678876545"
+        },
+        {
+            "idType": "SHOPIFY_ORD_ID",
+            "idValue": "4444992255"
+        }
+    ],
+    "paymentPreferences": [
+        {
+            "paymentMethodTypeId": "EXT_SHOP_OTHR_GTWAY",
+            "maxAmount": "25000",
+            "statusId": "PAYMENT_AUTHORIZED",
+            "paymentMode": "",
+            "cardName": "",
+            "manualAuthCode": null,
+            "manualRefNum": "45545454545454",
+            "createdByUserLogin": "nishtha.jain",
+            "orderId": "100783"
+        }
+    ],
+    "shipGroups": [
+        {
+            "facilityId": "NEW_ERA_HARAJUKU",
```

Comment:
Should we always expect facilityId i.e. OMS internal Id for the location for order creation?
As a generic API design rule, we can add facilityId and/or facilityExternalId, but if missing, do we return error and not create order?

Reply:
the createOrder service in OMS will expect fully resolved data. 

the createSalesOrder service is Shopify Integration application service. It has to take responsibility to resolve the data to ensure predictable results including graceful degradation.